src/components/Search/SearchContentPanel.tsx

```diff
@@ -0,0 +1,162 @@
+                        leave="transition ease-in duration-150"
+                        leaveFrom="opacity-100"
+                        leaveTo="opacity-0 translate-y-1"
+                    >
+                        <PopoverPanel
+                            className={classNames(
+                                'no-scrollbar absolute left-0 top-10 z-10 flex max-h-[275px] flex-col gap-2 overflow-y-auto rounded-lg bg-lightBottom py-3 text-medium shadow-popover dark:border dark:border-line dark:bg-darkBottom dark:shadow-none md:max-h-[370px]',
+                                popoverClassName,
+                            )}
+                        >
+                            {optionRenderer(data[0], -1, close)}
+                            {data.map((item, i) => optionRenderer(item, i, close))}
+                        </PopoverPanel>
+                    </Transition>
+                </>
+            )}
+        </Popover>
+    );
+}
+
+export function SearchContentPanel<T, F>({
+    showFilter = true,
+    isLoading,
+    filterProps,
+    placeholder,
+    itemRenderer,
+    onSearch,
+    onSelected,
+    listKey,
+    isSelected,
+}: SearchContentPanelProps<T, F>) {
+    const [filterIndex, setFilterIndex] = useState<number>(-1);
+    const [searchText, setSearchText] = useState('');
+    const listRef = useRef<HTMLDivElement>(null);
+
+    const debouncedSearch = useDebounce(searchText, 300);
+
+    const { value, loading } = useAsync(async () => {
+        if (isLoading) return [];
+        return await onSearch(debouncedSearch, filterProps.data[filterIndex]);
+    }, [debouncedSearch, filterIndex, isLoading]);
+
+    const selectedIndex = value?.findIndex((item) => isSelected?.(item)) ?? -1;
+
+    useEffect(() => {
+        // Scroll to the first selected item
+        if (selectedIndex >= 0 && listRef.current) {
+            const selectedEl = listRef.current.children[selectedIndex];
+            selectedEl?.scrollIntoView({ block: 'center' });
+        }
+    }, [value, selectedIndex]);
+
+    return (
+        <div className="h-full w-full">
+            <div className="flex items-center gap-2.5">
+                {showFilter ? (
+                    <FilterPopover {...filterProps} selected={filterIndex} onSelected={setFilterIndex} />
+                ) : null}
+                <div className="flex-1 rounded-lg !bg-lightBg">
+                    <SearchInput
+                        placeholder={placeholder}
+                        className="!focus:border-highlight rounded-lg !border border-transparent !py-1.5 px-3 transition-all"
+                        value={searchText}
+                        onChange={(event) => setSearchText(event.currentTarget.value)}
+                        onClear={() => setSearchText('')}
+                    />
+                </div>
+            </div>
+            <div ref={listRef} className="no-scrollbar mt-2 h-[calc(100%_-_44px)] overflow-y-auto">
```

Comment:
You might don't need `calc()`
```suggestion
            <div ref={listRef} className="no-scrollbar mt-2 min-h-0 flex-grow overflow-y-auto">
```